flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/core/testutils/RunnableWithException.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.core.testutils;
+
+@FunctionalInterface
+public interface RunnableWithException {
+
+    /**
+     * The work method.
+     *
+     * @throws Exception Exceptions may be thrown.
+     */
+    void run() throws Exception;
+}
```

Comment:
Why aren't you copying the definition from `ThrowingRunnable`?

Reply:
Because flink-core is not include in CheckedThread's module(flink-test-utils-junit), or It should be said that all Flink related dependencies are not included in it

Reply:
I'm aware that we can't use the implementation from flink-core; what I was asking is why this class, which has the exact same requirements as the ThrowingRunnable, is implemented differently.

Reply:
In fact, I copied the code from `org.apache.flink.util.function.RunnableWithException`, but your suggestion should be more reasonable. I will copy `org.apache.flink.util.function.ThrowingRunnable` directly.